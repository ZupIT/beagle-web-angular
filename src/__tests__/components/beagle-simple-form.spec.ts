/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, ComponentFixture, fakeAsync } from '@angular/core/testing'
import { FormsModule } from '@angular/forms'
import { BeagleSimpleFormComponent }
  from '../../components/beagle-simple-form/beagle-simple-form.component'

let component: BeagleSimpleFormComponent
let fixture: ComponentFixture<BeagleSimpleFormComponent>


describe('BeagleSimpleFormComponent', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        BeagleSimpleFormComponent,
      ],
      imports: [
        FormsModule,
      ],
    }).compileComponents()

    fixture = TestBed.createComponent(BeagleSimpleFormComponent)
    component = fixture.componentInstance
    component.onSubmit = jest.fn()
  }))

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should trigger on Submit', fakeAsync(() => {
    spyOn(component, 'handleSubmit').and.callThrough()
    component.handleSubmit(new Event('submit', {}))
    expect(component.onSubmit).toBeCalled()
  }))

})
