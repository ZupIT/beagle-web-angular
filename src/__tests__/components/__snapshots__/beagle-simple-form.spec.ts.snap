// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleSimpleFormComponent should match snapshot 1`] = `
BeagleSimpleFormComponent {
  "onSubmit": [MockFunction],
}
`;
